nsxt/resource_nsxt_policy_ipsec_vpn_session.go

```diff
@@ -0,0 +1,482 @@
+		Description: "For policy-based IPsec VPNs, a security policy specifies as its action the VPN tunnel to be used for transit traffic that meets the policy’s match criteria.",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"sources": {
+					Type:        schema.TypeSet,
+					Description: "List of local subnets. Specifying no value is interpreted as 0.0.0.0/0.",
+					Elem: &schema.Schema{
+						Type:         schema.TypeString,
+						ValidateFunc: validateCidr(),
+					},
+					Required: true,
+				},
+				"destinations": {
+					Type:        schema.TypeSet,
+					Description: "List of remote subnets used in policy-based L3Vpn.",
+					Elem: &schema.Schema{
+						Type:         schema.TypeString,
+						ValidateFunc: validateCidr(),
+					},
+					Required: true,
+				},
+				"action": {
+					Type:         schema.TypeString,
+					Description:  "PROTECT - Protect rules are defined per policy based IPSec VPN session. BYPASS - Bypass rules are defined per IPSec VPN service and affects all policy based IPSec VPN sessions. Bypass rules are prioritized over protect rules.",
+					Default:      model.IPSecVpnRule_ACTION_PROTECT,
+					Optional:     true,
+					ValidateFunc: validation.StringInSlice(IPSecRulesActionValues, false),
+				},
+			},
+		},
+	}
+}
+
+func getIPSecVPNRulesFromSchema(d *schema.ResourceData) []model.IPSecVpnRule {
+	rules := d.Get("rule").([]interface{})
+	var ruleList []model.IPSecVpnRule
+	for _, rule := range rules {
+		data := rule.(map[string]interface{})
+		action := data["action"].(string)
+		log.Println(action)
+		sourceRanges := interface2StringList(data["sources"].(*schema.Set).List())
+		destinationRanges := interface2StringList(data["destinations"].(*schema.Set).List())
+
+		/// Source Subnets
+
+		SourceIPSecVpnSubnetList := make([]model.IPSecVpnSubnet, 0)
+		if len(sourceRanges) > 0 {
+			for _, element := range sourceRanges {
+				subnet := element
+				log.Println(subnet)
+				IPSecVpnSubnet := model.IPSecVpnSubnet{
+					Subnet: &subnet,
+				}
+				SourceIPSecVpnSubnetList = append(SourceIPSecVpnSubnetList, IPSecVpnSubnet)
+			}
+		}
+
+		/// Destination Subnets
+		DestinationIPSecVpnSubnetList := make([]model.IPSecVpnSubnet, 0)
+		if len(destinationRanges) > 0 {
+			for _, element := range destinationRanges {
+				subnet := element
+				log.Println(subnet)
+				IPSecVpnSubnet := model.IPSecVpnSubnet{
+					Subnet: &subnet,
+				}
+				DestinationIPSecVpnSubnetList = append(DestinationIPSecVpnSubnetList, IPSecVpnSubnet)
+			}
+		}
+
+		ruleID := newUUID()
+		elem := model.IPSecVpnRule{
+			Action:       &action,
+			Sources:      SourceIPSecVpnSubnetList,
+			Destinations: DestinationIPSecVpnSubnetList,
+			UniqueId:     &ruleID,
+			Id:           &ruleID,
+		}
+		ruleList = append(ruleList, elem)
+	}
+	return ruleList
+}
+
+func resourceNsxtPolicyIPSecVpnSessionCreate(d *schema.ResourceData, m interface{}) error {
+
+	Tier0ID := d.Get("tier0_id").(string)
```

Comment:
according to coding conventions, local vars should start with lowercase, same below